src/platform/user/profile/vap-svc/actions/transactions.js

```diff
@@ -203,12 +203,14 @@ export function createTransaction(
         transaction,
       });
     } catch (error) {
+      const [firstError = {}] = error.errors ?? [];
+      const { code = 'code', title = 'title', detail = 'detail' } = firstError;
       const profileSection = analyticsSectionName || 'unknown-profile-section';
       recordEvent({
         event: 'profile-edit-failure',
         'profile-action': 'save-failure',
         'profile-section': profileSection,
-        'error-key': `transaction-creation-error-${profileSection}-save-failure`,
+        'error-key': `tx-creation-error-${profileSection}-${code}-${title}-${detail}`,
```

Comment:
I actually wrote a function that'd take these three vars and make a string that didn't include undefined values but that really felt like overkill.

```
function buildString(...args) {
	let result = args.reduce((string, arg) => {
		if (!!arg) {
			string += `${arg}-`
		}
		return string
	}, '')
	if (result.length) {
		return result.substr(0, result.length - 1)
	}
	return result
}
```